index.js

```diff
@@ -55,6 +55,73 @@ class Vector3 {
 	}
 }
 
+class STLMeasures {
+  constructor() {
+  	this.volume = 0;
+  	this.area = 0;
+  	this.minx = Infinity;
+  	this.maxx = -Infinity;
+  	this.miny = Infinity;
+  	this.maxy = -Infinity;
+  	this.minz = Infinity;
+  	this.maxz = -Infinity;
+    this.xCenter = 0,
+    this.yCenter = 0,
+    this.zCenter = 0;
+  }
+
+  addTriangle(triangle) {
+    let currentVolume = _triangleVolume(triangle);
+ 		this.volume += currentVolume;
+
+		const ab = triangle[1].clone().sub(triangle[0]);
+		const ac = triangle[2].clone().sub(triangle[0]);
+
+		this.area +=
+			ab
+				.clone()
+				.cross(ac)
+				.length() / 2;
+
+		const tminx = Math.min(triangle[0].x, triangle[1].x, triangle[2].x);
+		this.minx = tminx < this.minx ? tminx : this.minx;
+		const tmaxx = Math.max(triangle[0].x, triangle[1].x, triangle[2].x);
+		this.maxx = tmaxx > this.maxx ? tmaxx : this.maxx;
+
+		const tminy = Math.min(triangle[0].y, triangle[1].y, triangle[2].y);
+		this.miny = tminy < this.miny ? tminy : this.miny;
+		const tmaxy = Math.max(triangle[0].y, triangle[1].y, triangle[2].y);
+		this.maxy = tmaxy > this.maxy ? tmaxy : this.maxy;
+
+		const tminz = Math.min(triangle[0].z, triangle[1].z, triangle[2].z);
+		this.minz = tminz < this.minz ? tminz : this.minz;
+		const tmaxz = Math.max(triangle[0].z, triangle[1].z, triangle[2].z);
+		this.maxz = tmaxz > this.maxz ? tmaxz : this.maxz;
+
+    // Center of Mass calculation
+    // adapted from c++ at: https://stackoverflow.com/a/2085502/6482703
+    this.xCenter += ((triangle[0].x + triangle[1].x + triangle[2].x) / 4) * currentVolume;
+    this.yCenter += ((triangle[0].y + triangle[1].y + triangle[2].y) / 4) * currentVolume;
+    this.zCenter += ((triangle[0].z + triangle[1].z + triangle[2].z) / 4) * currentVolume;
+  }
+
+  finalize() {
+  	const volumeTotal = Math.abs(this.volume) / 1000;
+
+    this.xCenter /= this.volume;
+    this.yCenter /= this.volume;
+    this.zCenter /= this.volume;
+
+  	return {
+  		volume: volumeTotal, // cubic cm
+  		weight: volumeTotal * 1.04, // gm
+  		boundingBox: [this.maxx - this.minx, this.maxy - this.miny, this.maxz - this.minz],
+  		area: this.area,
+      centerOfMass: [this.xCenter, this.yCenter, this.zCenter],
```

Comment:
indention :-)

Reply:
Ah, tabs, I'm not used to them at all. It's fixed now I think.